test/common/access_log/access_log_impl_test.cc

```diff
@@ -1132,6 +1134,155 @@ name: envoy.file_access_log
   log->log(&request_headers_, &response_headers_, &response_trailers_, stream_info_);
 }
 
+class TestHeaderFilterFactory : public ExtensionFilterFactory {
+public:
+  ~TestHeaderFilterFactory() override = default;
+
+  FilterPtr createFilter(const envoy::config::filter::accesslog::v2::ExtensionFilter& config,
+                         Runtime::Loader&, Runtime::RandomGenerator&) override {
+    auto factory_config = Config::Utility::translateToFactoryConfig(
+        config, Envoy::ProtobufMessage::getNullValidationVisitor(), *this);
+    const auto& header_config =
+        MessageUtil::downcastAndValidate<const envoy::config::filter::accesslog::v2::HeaderFilter&>(
+            *factory_config);
+    return FilterPtr{new HeaderFilter(header_config)};
+  }
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return std::make_unique<envoy::config::filter::accesslog::v2::HeaderFilter>();
+  }
+
+  std::string name() const override { return "test_header_filter"; }
+};
+
+TEST_F(AccessLogImplTest, TestHeaderFilterPresence) {
```

Comment:
Nice test to show that existing filters in the core codebase can be registered with this mechanism. I was also wondering what @auni53 was wondering here https://github.com/envoyproxy/envoy/pull/7435/files#r302109914. Is this test added for completeness, or because there might be desire to do this with core codebase filters in the future?

Reply:
I would say that this test is added for completeness and to ensure that configuration of core codebase filters can be handled by extension filter. Unfortunately there doesn't seem to be a way to convert core filters to the same mechanism without changing configuration format.

Should I remove this test as duplicate?